include/vtk_utils.h

```diff
@@ -0,0 +1,188 @@
+  template <int dim, int spacedim>
+  void
+  read_vtk(const std::string            &vtk_filename,
+           Triangulation<dim, spacedim> &tria,
+           const bool                    cleanup = true);
+
+  /**
+   * @brief Read cell data (scalar or vector) from a VTK file and store it in
+   * the output vector.
+   *
+   * This function reads the specified cell data array (scalar or vector) from
+   * the given VTK file and stores it in the provided output vector. For vector
+   * data, all components are stored in row-major order (cell0_comp0,
+   * cell0_comp1, ..., cell1_comp0, ...).
+   *
+   * @param vtk_filename The name of the input VTK file.
+   * @param cell_data_name The name of the cell data array to read.
+   * @param output_vector The vector to store the cell data values.
+   */
+  void
+  read_cell_data(const std::string &vtk_filename,
+                 const std::string &cell_data_name,
+                 Vector<double>    &output_vector);
+
+  /**
+   * @brief Read point data from a VTK file and store it in the output vector.
+   *
+   * This function reads the specified point data array (scalar or vector) from
+   * the given VTK file and stores it in the provided output vector.
+   *
+   * @param vtk_filename The name of the input VTK file.
+   * @param point_data_name The name of the point data array to read.
+   * @param output_vector The vector to store the point data values.
+   */
+  void
+  read_point_data(const std::string &vtk_filename,
+                  const std::string &point_data_name,
+                  Vector<double>    &output_vector);
+
+  /**
+   * @brief Read a VTK mesh and all data fields into a DoFHandler and output
+   * vector.
+   *
+   * This function reads the mesh from the specified VTK file, populates the
+   * Triangulation associated to the given DoFHandler, and queries all cell and
+   * point data fields. For each data field, a suitable FESystem is constructed
+   * (using FE_DGQ for cell data and FE_Q for point data, with the correct
+   * number of components). DoFs are distributed and renumbered block-wise. All
+   * data is read into the output_vector, and the names of the fields are stored
+   * in data_names.
+   *
+   * @param vtk_filename The name of the input VTK file.
+   * @param dof_handler The DoFHandler to distribute DoFs on the mesh.
+   * @param output_vector The vector to store all data field values.
+   * @param data_names The vector to store the names of all data fields found in
+   * the VTK file.
+   */
+  template <int dim, int spacedim>
+  void
+  read_vtk(const std::string         &vtk_filename,
+           DoFHandler<dim, spacedim> &dof_handler,
+           Vector<double>            &output_vector,
+           std::vector<std::string>  &data_names);
+
+
+  // Custom comparator for Point<dim>.
+  template <int dim>
+  struct PointComparator
+  {
+    bool
+    operator()(const Point<dim> &p1, const Point<dim> &p2) const
+    {
+      // Compare lexicographically
```

Comment:
Can you make this a relative check? Something like

`const double tol = (p1.norm() + p2.norm())*.5e-7;`

When the domain is small, or the points are close to the origin 1e-10 seems a bit arbitrary

Reply:
I agree. So done, thanks.